src/natpmp.cpp

```diff
@@ -278,7 +277,8 @@ void natpmp::try_next_mapping(int const i)
 
 	auto const m = std::find_if(
 		m_mappings.begin(), m_mappings.end()
-		, [] (mapping_t const& ma) { return ma.act != mapping_t::action::none; });
+		, [] (mapping_t const& ma) { return ma.act != mapping_t::action::none &&
```

Comment:
I haven't dug into this problem or analyzed the code and the details you posted in the issue. That's why I'm asking, if you understand the problem it would save me some time if you could explain it.